sdk/keyvault/azure-security-keyvault-certificates/test/ut/certificate_client_test.cpp

```diff
@@ -572,7 +572,27 @@ TEST_F(KeyVaultCertificateClientTest, BackupRestoreCertificate)
     auto result = response.PollUntilDone(m_defaultWait);
     EXPECT_EQ(result.Value.Name(), certificateName);
     client.PurgeDeletedCertificate(certificateName);
-    std::this_thread::sleep_for(m_defaultWait);
+    TestSleep(m_defaultWait);
+  }
+  {
+    int retries = 15;
+    // before we restore we need to ensure the certificate is purged.
+    // since we have no visibility into the purge status we will just wait and check for a few
+    // seconds
+    while (retries > 0)
+    {
+      try
+      {
+        client.GetDeletedCertificate(certificateName);
+      }
+      catch (Azure::Core::RequestFailedException const&)
+      {
+        std::cout << std::endl << "- Certificate is gone";
+        break;
+      }
+      TestSleep(m_defaultWait);
+      retries--;
```

Comment:
I believe that the way this is written, the test passes after 15 iterations.

Is that your intent?

Reply:
it will wait for purge, then the purge fails the restore will fail

Reply:
And the purge is guaranteed to fail within a defined time less than an hour?

My primary concern here is tying up a pipeline for 2 hours because of the test functionally hanging because the service lost its marbles.